src/Common/Notifications/Notifications.php

```diff
@@ -0,0 +1,276 @@
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function get_api_url() {
+		$api = defined( 'TEC_COMMON_IAN_API_URL' ) ? TEC_COMMON_IAN_API_URL : 'https://ian.stellarwp.com/feed/stellar/tec/plugins.json';
+
+		/**
+		 * Filter the API URL for the In-App Notifications.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $api The API URL for the In-App Notifications.
+		 */
+		$api = apply_filters( 'tec_common_ian_api_url', $api, $this );
+
+		return $api;
+	}
+
+	/**
+	 * Define which pages will show the notification icon.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function is_ian_page() {
+		$screen = get_current_screen();
+
+		$allowed = [ 'tribe_events', 'edit-tribe_events', 'tribe_events_page_tec-events-settings' ];
+
+		/**
+		 * Filter the allowed pages for the Notifications icon.
+		 *
+		 * @since TBD
+		 *
+		 * @param array<string> $allowed The allowed pages for the Notifications icon.
+		 */
+		$allowed = apply_filters( 'tec_common_ian_allowed_pages', $allowed );
+
+		if ( in_array( $screen->id, $allowed, true ) ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Register the plugins that support In-App Notifications.
+	 *
+	 * @since TBD
+	 *
+	 * @return array<string> The slugs for plugins that support IAN.
+	 */
+	public function get_plugins() {
+		$plugins = [ 'the-events-calendar', 'event-tickets' ];
+
+		/**
+		 * Filter the plugin slugs for the In-App Notifications.
+		 *
+		 * @since TBD
+		 *
+		 * @param array<string> $slugs The slugs for plugins that support IAN.
+		 */
+		return apply_filters( 'tec_common_ian_slugs', $plugins );
+	}
+
+	/**
+	 * Show our notification icon.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $slug The plugin slug for IAN.
+	 *
+	 * @return void
+	 */
+	public function show_icon( $slug ): void {
+		if ( ! in_array( $slug, $this->get_plugins(), true ) || ! $this->is_ian_page() ) {
+			return;
+		}
+
+		/**
+		 * Filter allowing disabling of the Notifications icon by returning false.
+		 *
+		 * @since TBD
+		 *
+		 * @param bool $show Whether to show the icon or not.
+		 */
+		$show = (bool) apply_filters( 'tec_common_ian_show_icon', true, $slug );
+
+		if ( ! $show ) {
+			return;
+		}
+
+		$template = new Template();
+		$template->render_icon( [ 'slug' => $slug ], true );
+	}
+
+	/**
+	 * Optin to IAN notifications.
+	 *
+	 * @since TBD
+	 */
+	public function opt_in() {
+		if ( wp_verify_nonce( tec_get_request_var( 'nonce' ), 'common_ian_nonce' ) ) {
+
+			tribe_update_option( 'ian-notifications-opt-in', 1 );
+
+			wp_send_json_success( esc_html__( 'Notifications opt-in successful', 'tribe-common' ), 200 );
+		} else {
+			wp_send_json_error( esc_html__( 'Invalid nonce', 'tribe-common' ), 403 );
+		}
+	}
+
+	/**
+	 * Get the IAN notifications.
+	 *
+	 * @since TBD
+	 */
+	public function get_feed() {
+		if ( ! wp_verify_nonce( tec_get_request_var( 'nonce' ), 'common_ian_nonce' ) ) {
+			wp_send_json_error( esc_html__( 'Invalid nonce', 'tribe-common' ), 403 );
```

Comment:
To help you with testing, you will need to add a `return` after every wp_send_json which is bailing early.

Thats because you will need to mock these functions in order to test them to avoid their exit.

I will skip making this comment everywhere its applicable :D